doc/dev/adapters.md

```diff
@@ -1,15 +1,699 @@
+
+After `builder` object instantiation, Mixer configures the `builder` object by invoking various Template specific
+`HandlerBuilder `interface methods (example `SetMetricTypes`, `SetQuotaTypes` for 'metric' and 'quota' named Templates.)
+and passing a map of string-to-`Type `struct. The string key and the value `Type `represents the name of the instance as
+configured by the operator and the shape of the `Instance` object the adapter would receive during request time.
+
+Given the above sample operator's handler configuration and 'metric' Template shows in above examples, the below examples
+shows the configuration-time call values:
+
+![flow: example attr to types](./img/example%20attr%20to%20instance.svg)
+
+During request time, every `Instance` object dispatch to the adapter has a '`Name`' field. Adapter implementation should
+use the value of the `Name` field to lookup the shape description for the `Instance` object from the map of instance
+name(string)->`Type` that was passed during configuration time through the `builder` object.
+
+Once Mixer has called into various template-specific Set.*Types methods,
+
+Mixer calls the `SetAdapterConfig` method on the `builder`, and once done then Mixer calls the `Validate` followed by the
+`Build` method. `SetAdapterConfig` gives the builder the adapter specific configuration, `Validate` allows builder to
+validate the operator configuration based on the the provided Template specific `Types` and the Adapter specific configuration.
+
+**Instantiating `handler`**
+
+Once `builder` is validated, Mixer calls its `Build` method, which returns a `handler` object which Mixer invokes during
+request processing. The `handler `instance constructed must implement all the `Handler` interfaces (runtime request
+serving interfaces) for all the templates the adapter has registered for. If the returned handler fails to implement the
+required interface for the adapter's supported templates, Mixer reports an error and doesn't serve runtime traffic to
+the particular handler.
+
+*NOTE*: In the `Build` method, adapters must do all the bootstrapping work (example establishing connection with backend
+system, initializing cache and more) that they need to start receiving data during request time.
+
+**Closing `handler`**
+
+When a handler is no longer useful, Mixer calls it close method. In the `Close` method an adapter is expected to release
+all the allocated resources and close all remote connections to the backends if it has any.
+
+### Request-time
+
+During this time Mixer dispatches the `instance` objects to the adapter based on the routing rules that operator has
+configured. Mixer does this by invoking the Handle.* functions on the `handler` object.
+
+Given the above example operator's config (instance, action, handler configuration) and
+['metric' Template](#heading=h.ee6dn8otn4o0), the following examples shows the request-time `Instance` objects created
```

Comment:
Template -> template
And move "template" out of the hyperlink

Reply:
D